client/app/bundles/common/DashboardPage.tsx

```diff
@@ -0,0 +1,127 @@
+import { useMemo, useState } from 'react';
+import { defineMessages } from 'react-intl';
+import { Navigate } from 'react-router-dom';
+import { ArrowForward } from '@mui/icons-material';
+import { Avatar, Typography } from '@mui/material';
+import moment from 'moment';
+
+import SearchField from 'lib/components/core/fields/SearchField';
+import Page from 'lib/components/core/layouts/Page';
+import Link from 'lib/components/core/Link';
+import { useAppContext } from 'lib/containers/AppContainer';
+import useTranslation from 'lib/hooks/useTranslation';
+
+const translations = defineMessages({
+  searchCourses: {
+    id: 'lib.components.navigation.CourseSwitcherPopupMenu.searchCourses',
+    defaultMessage: 'Search your courses',
+  },
+  jumpBackIn: {
+    id: 'lib.components.navigation.CourseSwitcherPopupMenu.jumpBackIn',
+    defaultMessage: 'Jump back in',
+  },
+  lastAccessed: {
+    id: 'lib.components.navigation.CourseSwitcherPopupMenu.lastAccessed',
+    defaultMessage: 'Last accessed {at}',
+  },
+  noCoursesMatch: {
+    id: 'lib.components.navigation.CourseSwitcherPopupMenu.noCoursesMatch',
+    defaultMessage: "Oops, no courses matched ''{keyword}''.",
+  },
+});
+
+const DashboardPage = (): JSX.Element => {
+  const { courses } = useAppContext();
+
+  const { t } = useTranslation();
+
+  const [filterKeyword, setFilterKeyword] = useState('');
```

Comment:
Reuse useItems for filtering and sorting?

Reply:
I don't think it's useful in this case. There's no pagination in this page.

Reply:
It is not necessary to use the pagination feature, you can just pass the whole length of items to the hook.

Reply:
You will start to have repeated code in the codebase (ie DashboardPage, DayView, etc)